wandb/sdk/wandb_init.py

```diff
@@ -695,10 +695,17 @@ def _attach(
 
     manager = _wl._get_manager()
     if manager:
-        manager._inform_attach(attach_id=attach_id)
+        response = manager._inform_attach(attach_id=attach_id)
 
     settings: Settings = copy.copy(_wl._settings)
-    settings.update(run_id=attach_id, source=Source.INIT)
+    settings.update(
+        {
+            "run_id": attach_id,
+            "_start_time": response["_start_time"],
```

Comment:
will this (and `_start_datetime`) always be set by the time attach happens? just thinking about potential validation errors if these are None's or something.

Reply:
`attach` happens on a run that already have been created before (so expecting it to be not a `None` value), the only case where this would be a `None` if the original run has `None` values for these... do you think we need to add some assertions here?

Reply:
sounds like a pathological case, so let's rely on settings throwing validation errors instead.